cli/cppcheckexecutor.h

```diff
@@ -79,7 +79,7 @@ class CppCheckExecutor : public ErrorLogger {
     /** xml output of errors */
     void reportErr(const ErrorMessage &msg) override;
 
-    void reportProgress(const std::string &filename, const char stage[], const std::size_t value) override;
+    void reportProgress(const std::string &filename, const char stage[], std::size_t value) override;
```

Comment:
I don't have a very strong opinion. But I don't see how it would hurt to keep these const.

Reply:
Well they have no effect and we also don't have the definition and declaration in sync. In some cases it might even be misleading.  It also exposes bugs as we are not properly handling this code.

Reply:
The point is to have `const` only in the implementation (where it matters), not in the declaration, right? So we don't actually lose anything here, since top-level `const` has no effect in a function declaration.

Reply:
it has no effect on the function interface. but it does have an effect in the function body.

> The point is to have const only in the implementation (where it matters), not in the declaration, right?

In my opinion it would be a bad idea to have different const-qualifiers in the declaration and implementation.

Reply:
It has no effect. there's also cases where the header and source were not in sync and nothing will tell you. It actually doesn't matter at all. It just makes things "cleaner" - and exposes Cppcheck bugs :)

Reply:
> It has no effect. 

As I said it has an effect in the function body.

    void foo(const int x) {
        x=3; // <- not allowed because of const
    } 

I do not have a strong opinion if we should use const for parameters passed by value. But in my opinion we should not have different constness in the declaration and implementation. If the implementation uses const then I feel the declaration should too.

> there's also cases where the header and source were not in sync and nothing will tell you. 

Are you sure that this is explicitly permitted by the language standard? Can it be a compiler bug?

Reply:
> Well.. it does have an effect in the function body.

That wasn't changed.

> Are you sure that this is explicitly permitted by the language standard? Can it be a compiler bug?

Yes, it is allowed. No, it is not a bug. That's what this check is about: https://clang.llvm.org/extra/clang-tidy/checks/readability/avoid-const-params-in-decls.html

The following are the same as the top-level `const` has no effect:
```cpp
void f(const int);
void f(int);
```

Reply:
well.. I guess the language standard permits it then but I see no reference..

imho.. having different const in the declaration and implementation seems weird to me. And nobody seems to disagree.